### PR TITLE
BFB porting droplet activation to scream

### PR DESCRIPTION
**Pull request by xyuan**
BFB porting droplet activation to scream

BFB

### CODE REVIEW

**Comment by xyuan** [comment]
sizeof(Real) 8 avx  packsize 1 compiler GCC FPE off #threads 1
Comparing with -t8 at tol 0.0e+00
--- checking case # 1, it = 1/1 ---
--- checking case # 2, it = 1/6 ---
--- checking case # 2, it = 2/6 ---
--- checking case # 2, it = 3/6 ---
--- checking case # 2, it = 4/6 ---
--- checking case # 2, it = 5/6 ---
--- checking case # 2, it = 6/6 ---
--- checking case # 3, it = 1/1 ---
--- checking case # 4, it = 1/6 ---
--- checking case # 4, it = 2/6 ---
--- checking case # 4, it = 3/6 ---
--- checking case # 4, it = 4/6 ---
--- checking case # 4, it = 5/6 ---
--- checking case # 4, it = 6/6 ---

and p3_tests
sizeof(Real) 8 avx  packsize 1 compiler GCC FPE off #threads 1
===============================================================================
All tests passed (78040 assertions in 35 test cases)

File: components/scream/src/physics/p3/p3_functions_droplet_activation_impl.hpp [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+   constexpr Scalar cons7  = C::CONS7;
+   constexpr Scalar nccnst = C::NCCNST;
+   constexpr Scalar rv     = C::RH2O;
+   constexpr Scalar inv_cp = C::INV_CP;
+
+   const auto sup_gt_small = sup > sp(1.e-6);
+   const auto sup_if_log   = sup_gt_small && log_predictNc;
+   const auto sup_if_not_log = sup_gt_small && !log_predictNc;
+
+   Spack dum{0.}, dumqvs{0.}, dqsdt{0.}, ab{0.};
```
Comment (jgfouca) [inline components/scream/src/physics/p3/p3_functions_droplet_activation_impl.hpp]: Use integer literals for integers.

File: components/scream/src/physics/p3/p3_functions_droplet_activation_impl.hpp [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+     qcnuc.set(sup_if_log,
+               ncnuc*cons7);
+  }
+
+  // for specified Nc, make sure droplets are present if conditions are supersaturated
+  // this is not applied at the first time step, since saturation adjustment is applied at the first step
+
+  if (sup_if_not_log.any()) {
+      dum   = nccnst*inv_rho*cons7-qc;
+      dum   = max(sp(0.),dum);
```
Comment (jgfouca) [inline components/scream/src/physics/p3/p3_functions_droplet_activation_impl.hpp]: Use integer literals for integers.

File: components/scream/src/physics/p3/p3_functions_droplet_activation_impl.hpp [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+
+      dumqvs = qv_sat(temp,pres,false);
+
+      dqsdt = xxlv*dumqvs/(rv*temp*temp);
+
+      ab    = sp(1.) + dqsdt*xxlv*inv_cp;
```
Comment (jgfouca) [inline components/scream/src/physics/p3/p3_functions_droplet_activation_impl.hpp]: Use integer literals for integers.

**Comment by xyuan** [comment]
close this PR, and redo it based on the latest mater branch

**Pull request closed without merging**